domain/src/test/java/org/oppia/android/domain/topic/StoryProgressControllerTest.kt

```diff
@@ -115,6 +115,34 @@ class StoryProgressControllerTest {
     verifyRecordProgressSucceeded()
   }
 
+  @Test
+  fun testStoryProgressController_recordChapterAsInProgressSaved_isSuccessful() {
+    storyProgressController.recordChapterAsInProgressSaved(
+      profileId,
+      FRACTIONS_TOPIC_ID,
+      FRACTIONS_STORY_ID_0,
+      FRACTIONS_EXPLORATION_ID_0,
+      fakeOppiaClock.getCurrentTimeMs()
+    ).toLiveData().observeForever(mockRecordProgressObserver)
+    testCoroutineDispatchers.runCurrent()
+
+    verifyRecordProgressSucceeded()
+  }
+
+  @Test
+  fun testStoryProgressController_recordChapterAsInProgressNotSaved_isSuccessful() {
+    storyProgressController.recordChapterAsInProgressNotSaved(
+      profileId,
+      FRACTIONS_TOPIC_ID,
+      FRACTIONS_STORY_ID_0,
+      FRACTIONS_EXPLORATION_ID_0,
+      fakeOppiaClock.getCurrentTimeMs()
+    ).toLiveData().observeForever(mockRecordProgressObserver)
+    testCoroutineDispatchers.runCurrent()
+
+    verifyRecordProgressSucceeded()
+  }
```

Comment:
There are a lot of branches in recordChapterAsInProgressNotSaved. Please make sure to add more tests to cover all of the different scenarios in which that method runs to ensure it's well covered.

Reply:
added more tests